packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -1099,12 +1132,26 @@ export class BlockNoteEditor<
   }
 
   /**
-   * Gets a snapshot of the current selection.
+   * Gets a snapshot of the current selection. This contains all blocks (included nested blocks)
+   * that the selection spans across.
+   *
+   * If the selection starts / ends halfway through a block, the returned data will contain the entire block.
    */
   public getSelection(): Selection<BSchema, ISchema, SSchema> | undefined {
     return this.transact((tr) => getSelection(tr));
   }
 
+  /**
+   * Gets a snapshot of the current selection. This contains all blocks (included nested blocks)
+   * that the selection spans across.
+   *
+   * If the selection starts / ends halfway through a block, the returned block will be
+   * only the part of the block that is included in the selection.
+   */
+  public getSelectionCutBlocks() {
+    return this.transact((tr) => getSelectionCutBlocks(tr));
+  }
```

Comment:
I think the distinction here is selectionAround vs. selectionWithin

getSelection is getSelectionAround
getSelectionCutBlocks is getSelectionWithin

Reply:
I can see we'd like a better name, but I also don't find "Around" vs "Within" very intuitive tbh. Unless you've seen that pattern before?

Reply:
Prosemirror's replaceAround for one (it encompassing the whole node). 

Maybe it is just one getSelection, and it can return `blocks` vs `partialContent`?

Reply:
@matthewlipski also had good ideas about this with maybe 1 method that exposes both values; wdyt about that?

@matthewlipski can you write down your earlier thoughts?

Reply:
I would just have a single `getSelection`:

```ts
public getSelection(): {
    // Blocks spanned by selection, same as existing `getSelection`.
    blocks: Block<BSchema, I, S>[];
    // Fields returned by what's currently `getSelectionCutBlocks.
    cutBlocks: Block<BSchema, I, S>[];
    blockCutAtStart: string | undefined;
    blockCutAtEnd: string | undefined;
} { ... }
```

This way we don't have to have 2 separate functions (which imo is confusing for consumers) while also not introducing any breaking changes.